clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -1839,6 +1839,41 @@ static void handleRestrictAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
                  RestrictAttr(S.Context, AL, DeallocE, DeallocPtrIdx));
 }
 
+static bool isSpanLikeType(const QualType &Ty) {
```

Comment:
Rather than this, what about a `checkSpanLikeType` where we actually do diagnostics?